pkg/jx/cmd/controller_pipelinerunner.go

```diff
@@ -180,11 +180,12 @@ func (o *ControllerPipelineRunnerOptions) startPipelineRun(w http.ResponseWriter
 		return
 	}
 
-	// lets change this to support new pipelineresource type that handles batches
-	if len(pj.Refs.Pulls) > 0 {
+	// Only if there is one Pull in Refs, it's a PR build so we are going to pass it
+	if len(pj.Refs.Pulls) == 1 {
```

Comment:
I'd would be tempted to create a function `isSinglePullRequest()`, even if it in the end just contains just one line. IMO such a method speaks more for itself than the current if statement + comment

Reply:
So this is just to make sure we are passing the pull request number along, right?

Reply:
The thing here is that, the `> 0` condition was being matched by both PullRequest anad Batch build types. This caused the batch build to incorrectly contain information about the first PullRequest it contained.